.github/workflows/build_and_test.yml

```diff
@@ -0,0 +1,279 @@
+name: build_and_test
+on: [ push ]
```

Comment:
Curious if we run into the same issue as https://github.com/neondatabase/neon/pull/1797 did, when the same set of tests with `on: [ push ]` got triggered twice when a person pushed into his branch and created the PR.

Ideally we want to keep the tests running for pushes, since otherwise people are forced to create PRs to see if the tests suite passes, but duplicate test run is definitely odd as well.